miio/g1vacuum.py

```diff
@@ -0,0 +1,312 @@
+import logging
+from typing import Any, Dict
+from enum import Enum
+import click
+from .click_common import EnumType, command, format_output
+from .miot_device import MiotDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+_MAPPING = {
+    # https://miot-spec.org/miot-spec-v2/instance?type=urn:miot-spec-v2:device:vacuum:0000A006:mijia-v1:1
+    "battery": {"siid": 3, "piid": 1},
+    "charge_state": {"siid": 3, "piid": 2},
+    "error": {"siid": 2, "piid": 2},
+    "state": {"siid": 2, "piid": 1},
+    "fan_speed": {"siid": 2, "piid": 6},
+    "operating_mode": {"siid": 2, "piid": 4},
+    "mop_state": {"siid": 16, "piid": 1},
+    "water_level": {"siid": 2, "piid": 5},
+    "brush_life_level": {"siid": 14, "piid": 1},
+    #  "brush_life_time": {"siid": 14, "piid": 2},
+    "brush_life_level2": {"siid": 15, "piid": 1},
+    #  "brush_life_time2": {"siid": 15, "piid": 2},
+    "filter_life_level": {"siid": 11, "piid": 1},
+    #  "filter_life_time": {"siid": 11, "piid": 2},
+    "clean_area": {"siid": 9, "piid": 1},
+    "clean_time": {"siid": 18, "piid": 5},
+    "total_clean_count": {"siid": 9, "piid": 5},
+    #   "total_clean_area": {"siid": 9, "piid": 3},
+    #  "dnd_enabled": {"siid": 12, "piid": 2},
+    #  "audio_volume": {"siid": 4, "piid": 2},
+    #  "direction_key": {"siid": 8, "piid": 1}
+}
+
+class ChargeState(Enum):
+    Not_charging = 0
+    Charging = 1
+    Charging_competely = 2
+
+class Error(Enum):
+    Left_wheel_error = 1
+    Right_wheel_error = 2
+    Cliff_error = 3
+    Low_battery_error = 4
+    Bump_error = 5
+    Main_brush_error = 6
+    Side_brush_error = 7
+    Fan_motor_error = 8
+    Dustbin_error = 9
+    Charging_error = 10
+    No_water_error = 11
+    Everything_is_ok = 0
+    Pick_up_error = 12
```

Comment:
I think it would be better to follow the format as used in `vacuumcontainers.py`, so avoiding enums and using a dictionary for this mapping. See https://github.com/rytilahti/python-miio/blob/master/miio/vacuumcontainers.py#L125